proposals/XXXX-count-where.md

```diff
@@ -0,0 +1,70 @@
+# `count(where:)`
+
+* Proposal: [SE-NNNN](NNNN-filename.md)
+* Authors: [Soroush Khanlou](https://github.com/khanlou)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+
+## Introduction
+
+While Swift's `Sequence` models brings a lot of niceties that we didn't have access to in Objective-C, like `map` and `filter`, there are other useful operations on sequences that the standard library doesn't support yet. One current missing operation is  `count(where:)`, which counts the number of elements in a `Sequence` that pass some test.
+
+Swift-evolution thread: [`count(where:)` on Sequence](https://forums.swift.org/t/count-where-on-sequence/11186)
+
+## Motivation
+
+Counting the number of objects that pass a test has a wide range of uses in many domains. However, Swift currently doesn't give its users a simple way to perform this operation. While the behavior can currently be approximated with a `filter` and a `count`, this approach creates an intermediate array which it immediately discards. This is a bit wasteful.
+
+    [1, 2, 3, -1, -2].filter({ $0 > 0 }).count // => 3
```

Comment:
You can, instead of adding the four spaces, wrap these in backticks (\`\`\`) and add syntax highlighting (\`\`\`swift on the first line)